### PR TITLE
arch:arm:dts: Add ast2700-sp7 initial device tree

### PR DESCRIPTION
**Pull request by rajaganeshr**
Adding initial device tree for SP7 dc-scm board based on AST2700

### CODE REVIEW

File: configs/ast2700_sp7_defconfig [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+CONFIG_DEBUG_UART=y
+CONFIG_SPI_ASPEED_DMA=y
+CONFIG_SYS_MEMTEST_START=0x403000000
+CONFIG_SYS_MEMTEST_END=0x403001000
+# CONFIG_SYS_MALLOC_CLEAR_ON_INIT is not set
+CONFIG_FIT=y
+CONFIG_FIT_SIGNATURE=y
+CONFIG_LEGACY_IMAGE_FORMAT=y
+CONFIG_USE_BOOTARGS=y
+CONFIG_BOOTARGS="console=ttyS12,115200n8 root=/dev/ram rw earlycon vmalloc=512MB"
```
Comment (supven01) [inline configs/ast2700_sp7_defconfig]: vmalloc may not be needed. Please check.

File: configs/ast2700_sp7_defconfig [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+CONFIG_SPI_DIRMAP=y
+CONFIG_SPI_ASPEED_SMC=y
+CONFIG_SYSRESET=y
+CONFIG_UFS=y
+CONFIG_CADENCE_UFS=y
+CONFIG_ASPEED_UFS=y
+# CONFIG_WATCHDOG_AUTOSTART is not set
+CONFIG_WDT=y
+# CONFIG_WDT_ASPEED is not set
+CONFIG_WDT_AST2600=y
```
Comment (supven01) [inline configs/ast2700_sp7_defconfig:123]: AST2600?

File: configs/ast2700_sp7_defconfig [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+CONFIG_SYS_PROMPT="sp7# "
```
Comment (supven01) [inline configs/ast2700_sp7_defconfig:124]: should it be malta#?
How do we separate malta from bahama?

File: configs/ast2700_sp7_defconfig (hunk 2 above)
Comment (rajaganeshr) [reply configs/ast2700_sp7_defconfig:123]: SDK is using same driver for both 2600 and 2700

Comment (supven01) [reply configs/ast2700_sp7_defconfig:123]: Ok.

File: configs/ast2700_sp7_defconfig (hunk 3 above)
Comment (rajaganeshr) [reply configs/ast2700_sp7_defconfig:124]: Right now, only added nodes for malta, but kept name common as sp7. Still exploring device tree selection in u-boot SPL.

File: configs/ast2700_sp7_defconfig (hunk 1 above)
Comment (rajaganeshr) [reply configs/ast2700_sp7_defconfig]: fixed in 2nd commit of this pull request

**Review by supven01 — APPROVED** [review]

**Pull request merged**